test/e2e/e2e.go

```diff
@@ -101,10 +101,17 @@ func setupProviderConfig() error {
 			return fmt.Errorf("Error building GCE/GKE provider: %v", err)
 		}
 
+		cs, err := framework.LoadClientset()
+		if err != nil {
+			glog.Fatal("Error loading client: ", err)
+		}
+		gceCloud.SetClientSet(cs)
```

Comment:
It seems to be set using a method "Initialize" in gce.go. It's the line "gce.client = clientBuilder.ClientOrDie("cloud-provider")".

Looks like in production code this is done separately from creating the gceCloud object itself because it is an interface method and is called for any of the many cloud objects. It is called here:
https://github.com/kubernetes/kubernetes/blob/6b6b1e5779449a63079c9ff400da752975fe8a06/cmd/cloud-controller-manager/app/controllermanager.go#L176

Additionally I'm not sure that the clientset is a "config" item. In production code the cloudconfig is generated from reading in a config file and the clientset doesn't really seem to fit into that paradigm.

Setting the clientset with a method (although not the same method) after it is created is consistent with how it works in the production code. I think adding it to the CloudConfig for the e2e tests would actually make it inconsistent with the production code. I am open to adding it to the CloudConfig if you still think it is the right move, just wanted to provide you with additional info before we make a decision.

Reply:
Ok that's fine then. Thanks for the detailed explanation!